src/main/java/spoon/support/reflect/code/CtJavaDocImpl.java

```diff
@@ -96,6 +99,75 @@ public String getShortDescription() {
 		}
 	}
 
+	/**
+	 * Parses the content string to split in two: the description and the Javadoc tags
+	 */
+	@Override
+	public <E extends CtComment> E setContent(String content) {
+		tags.clear();
+
+		// avoiding NPE later
+		if (content == null) {
+			content = "";
+		}
+
+		String longDescription = "";
+		String currentTagContent = "";
+		CtJavaDocTag.TagType currentTag = null;
+
+		String[] lines = cleanComment(content).split("\n|\r\n|\r");
+		boolean tagStarted = false;
+		for (String aLine : lines) {
+			String line = aLine.trim();
+			if (line.startsWith(CtJavaDocTag.JAVADOC_TAG_PREFIX)) {
+				tagStarted = true;
+				int endIndex = line.indexOf(' ');
+				if (endIndex == -1) {
+					endIndex = line.length();
+				}
+				defineCommentContent(currentTagContent, currentTag);
+
+				currentTag = CtJavaDocTag.TagType.tagFromName(line.substring(1, endIndex).toLowerCase());
+				if (endIndex == line.length()) {
+					currentTagContent = "";
+				} else {
+					currentTagContent = line.substring(endIndex + 1);
+				}
+			} else {
+				if (tagStarted == false) {
+					longDescription += CtComment.LINE_SEPARATOR + aLine;
+				} else {
+					currentTagContent += CtComment.LINE_SEPARATOR + aLine;
+				}
+			}
+		}
+		defineCommentContent(currentTagContent, currentTag);
+
+		// we cannot call super.setContent because it calls cleanComment (which has already been done above)
+		// and we don't want to clean the comment twice
+		String contentWithTags = longDescription.trim(); // trim is required for backward compatibility
+		getFactory().getEnvironment().getModelChangeListener().onObjectUpdate(this, CtRole.COMMENT_CONTENT, contentWithTags, this.content);
+		this.content = contentWithTags;
+
+		return (E) this;
+	}
+
+
+
+	/**
+	 * Define the content of the comment
+	 * @param comment the comment
+	 * @param tagContent the tagContent of the tag
+	 * @param tagType the tag type
+	 */
+	private void defineCommentContent(String tagContent, CtJavaDocTag.TagType tagType) {
+		if (tagType != null) {
+			CtJavaDocTag docTag = this.getFactory().Code().createJavaDocTag(tagContent, tagType);
+			this.addTag(docTag);
+		} else if (!tagContent.isEmpty()) {
+			this.setContent(tagContent.trim());
```

Comment:
Is it recursive call to setContent which calls defineCommentContent, which calls setContent ... ?
I have not checked it deeper I just do not understand it on first sight.

Reply:
fixed!